### PR TITLE
Added recipe for pycryptodome

### PR DESCRIPTION
**Pull request by pmlandwehr**

### CODE REVIEW

File: recipes/pycryptodome/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    - Crypto.SelfTest.PublicKey
+    - Crypto.SelfTest.Random
+    - Crypto.SelfTest.Signature
+    - Crypto.SelfTest.Util
+    - Crypto.Signature
+    - Crypto.Util
+
+about:
+  home: http://www.pycryptodome.org
+  license_file: {{ RECIPE_DIR }}/LICENSE.rst
```
Comment (jakirkham) [inline recipes/pycryptodome/meta.yaml]: Please submit a PR upstream to add `LICENSE.rst` to the `MANIFEST.in` for future distribution. Also please link that PR in a comment in the recipe above the `license_file` line. Thanks.

Comment (pmlandwehr) [reply recipes/pycryptodome/meta.yaml]: Double-checked the source tarball, and it turns out a license is already included. No need for a manifest.

File: recipes/pycryptodome/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - toolchain
+
+  run:
+    - python
```
Comment (isuruf) [inline recipes/pycryptodome/meta.yaml]: How about adding `gmp` here?

From the README,

> For faster public key operations, you should have GMP installed in your system

**Comment by pmlandwehr** [comment]
Is there a decent way to force conda to have pycrypto be absent from the environment when doing setup? Per the docs, these two libraries shouldn't be installed at the same time and I don't know a good way to wrangle that.

**Comment by jakirkham** [comment]
`pycrypto` shouldn't be in the environments that the building or testing is being done. It will only be in the `root` environment (because of `conda`), which is separate. So you should be ok.

File: recipes/pycryptodome/meta.yaml (hunk 2 above)
Comment (jakirkham) [reply recipes/pycryptodome/meta.yaml]: Note: Done below.

How can we test that they are working ok?

File: recipes/pycryptodome/meta.yaml [hunk 3]
```diff
@@ -26,6 +26,8 @@ requirements:
 
   run:
     - python
+    - gmp  # [not win]
+    - mpir  # [win]
```
Comment (isuruf) [inline recipes/pycryptodome/meta.yaml]: You don't need `mpir` here as they are vendoring a copy in windows.

Comment (pmlandwehr) [reply recipes/pycryptodome/meta.yaml]: `mpir` is in the wheel, but is it also in the source?

**Comment by jakirkham** [comment]
Friendly nudge. 😉

**Comment by nehaljwani** [comment]
@jakirkham 

I found out how to test this.

```
from Crypto.Math import _Numbers_gmp as NumbersGMP
```

For pycrytodomex, it is same. We just have to replace Crypto with Cryptodomex

This will fail if `libgmp.{so,dylib}` or `mpir.dll` is not found.

However, it uses ctypes' `load_library()` function and fails to pick it up from `$CONDA_PREFIX/lib`

On macOS, I was able to do: 
```
DYLD_LIBRARY_PATH=$CONDA_PREFIX/lib python -c 'from Crypto.Math import _Numbers_gmp as NumbersGMP'
```

On Windows, installing mpir suffices, as the .dll is found in `PATH`

**Comment by jakirkham** [comment]
So we just added `pycryptodomex` ( https://github.com/conda-forge/staged-recipes/pull/4260 ), which should satisfy this need. Given `pycryptodome` and `pycrypto` use the same import name and `pycryptodomex` is identical to `pycryptodome` (modulo the import name), I think we should close this one. Thoughts from others?

**Pull request closed without merging**